rust/protocol/src/storage/traits.rs

```diff
@@ -1,38 +1,73 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Traits defining several stores used throughout the Signal Protocol.
+
 use async_trait::async_trait;
 use uuid::Uuid;
 
-use crate::state::{PreKeyId, SignedPreKeyId};
 use crate::{
-    IdentityKey, IdentityKeyPair, PreKeyRecord, ProtocolAddress, Result, SenderKeyRecord,
-    SessionRecord, SignedPreKeyRecord,
+    address::ProtocolAddress,
+    error::Result,
+    sender_keys::SenderKeyRecord,
+    state::{PreKeyId, PreKeyRecord, SessionRecord, SignedPreKeyId, SignedPreKeyRecord},
+    IdentityKey, IdentityKeyPair,
 };
 
+/// Handle to FFI-provided context object.
+///
+/// This object is not manipulated in Rust, but is instead passed back to each FFI
+/// method invocation. This argument should just be [None] for all clients of the Rust-only API.
 pub type Context = Option<*mut std::ffi::c_void>;
 
+/// Each Signal message can be considered to have exactly two participants, a sender and receiver.
+///
+/// [IdentityKeyStore::is_trusted_identity] uses this to ensure the identity provided is configured
+/// for the appropriate role.
+///
+/// *TODO: consider moving this enum into [crate::protocol]?*
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub enum Direction {
+    /// We are in the context of sending a message.
     Sending,
+    /// We are in the context of receiving a message.
     Receiving,
 }
 
+/// Interface defining the identity store, which may be in-memory, on-disk, etc.
+///
+/// Signal clients usually use the identity store in a [TOFU] manner, but this is not at
+/// all guaranteed.
```

Comment:
I'd say it's not *required.* The trait has a "save" API and an "is trusted" requirement, but the implementing type could choose some strategy other than TOFU.

Reply:
That simplifies it a lot, thanks!